### ISSUE TITLE
Scenes, sources, filters vanishing into thin air

### ISSUE THREAD

**Issue by MicaLovesKPOP** [obsproject/obs-studio#10645]
### Operating System Info

Windows 11

### OBS Studio Version

30.1.2

### OBS Studio Log URL

https://obsproject.com/logs/9SQ8s1AZtw27PKPR

### Expected Behavior

Nothing (just OBS launching and loading my saved profiles as usual)

### Current Behavior

My Scenes, sources, filters and scene transitions vanished. My audio mixer sources are still there. My OBS Settings are still there.

It seems the .JSON (attached below) got practically reset when I launched OBS today, though it is still twice much larger in file size than a newly created one.
[Assetto Corsa.json](https://github.com/obsproject/obs-studio/files/15221562/Assetto.Corsa.json)

### Steps to Reproduce

1. Likely changing GPU and/or reinstalling Windows
2. Start OBS and let it load the .JSON

### Anything else we should know?

I've ran into this issue twice before, the first time was in 2014. Every time I stop using OBS and then years later return to OBS, only to find out that this honestly terrible issue still persists. I hope by personally reporting it, it can finally get squashed now.

**Comment by RytoEX** [comment]
Did you have other scene collection files present in your userdata when this happened? What do you mean "re-installing Windows"? Did you back up your OBS settings, reinstall Windows, then reinstall OBS, and copy your OBS settings back after that?

We've had rare reports of this, and we have tried many times to reproduce this, but we have never been able to see it occur ourselves.

> though it is still twice much larger in file size than a newly created one.

It's twice as large as a newly created scene collection JSON because the default OBS scene collection (Untitled, Untitled.json) has two default audio sources: Desktop Audio and Mic/Auxiliary Audio. Newly created scene collections in OBS do not. In any case, it's not relevant to the core issue here.

If you have any backed up userdata (scene collection, ini/config files), it might be helpful for us to examine it.

**Comment by Mythotical** [comment]
I'm having this exact issue. I have to re-add my sources all the time. Scenes stay just sources vanish into thin air. I'm doing a complete reinstall to see if that fixes the issue.

**Comment by Fenrirthviti** [comment]
Just as a note, a reinstall rarely, if ever, will solve issues with OBS unless there a specific missing files that are required for OBS to run.

I understand the desire to want to fix the issue as fast as possible, but understanding what happened and getting us logs/collections/etc. as requested above is immensely helpful, as we have been trying to replicate these reported failures for over 6 years now, and have failed to do so. We need more information on what actions people are taking, or how they are causing this to happen.

**Comment by Mythotical** [comment]
Here are the files from me.
[2024-06-03 07-49-44.txt](https://github.com/user-attachments/files/15811991/2024-06-03.07-49-44.txt)
[2024-06-12 18-25-20.txt](https://github.com/user-attachments/files/15811992/2024-06-12.18-25-20.txt)

The first log is when it first happened after upgrading to Beta 2 since I'm part of the Enhanced Broadcasting beta with Twitch. The second is after a reinstall just now.

[2024-06-03 07-49-44.csv.gz](https://github.com/user-attachments/files/15812002/2024-06-03.07-49-44.csv.gz)
[2024-06-12 18-25-20.csv.gz](https://github.com/user-attachments/files/15812003/2024-06-12.18-25-20.csv.gz)

There is my profiler data, the first one of course is when it first happened and the second is after reinstall just now.

**Comment by RytoEX** [comment]
The first log (2024-06-03 07-49-44.txt) is using the Twitch Enhanced Broadcasting beta client.
> OBS 30.1.2-enhanced-broadcasting-v22

Twitch's client runs in Portable Mode by default:
> 07:49:44.359: Portable mode: true

Portable Mode will load no scenes or userdata unless you pre-copied files to the application's local config folder. This looks like expected behavior in that log.

The second log (2024-06-12 18-25-20.txt) is using our beta client.
> OBS 30.2.0-beta1

You have also configured this to run in Portable Mode:
> 18:25:21.003: Portable mode: true

It looks like it loaded scene data, but no sources. We would need the scene collection JSON files to determine if this is correct or not.

The profiler data will not be helpful, so you don't need to provide it.

**Comment by Mythotical** [comment]
Here you go, here is the scene collection file.

[Twitch.json.txt](https://github.com/user-attachments/files/15812203/Twitch.json.txt)

**Comment by RytoEX** [comment]
> Here you go, here is the scene collection file.
> 
> [Twitch.json.txt](https://github.com/user-attachments/files/15812203/Twitch.json.txt)

That scene collection file only has the scenes and no sources, so the OBS log showing only scenes and no sources is "correct" in that regard. If there _were_ supposed to be sources/sceneitems here, then those are not reflected in the scene collection file. They could be in the backup scene collection file (the `.bak` file with the same name), but it may already have been overwritten.

**Comment by Mythotical** [comment]
Oh ok. I have a new issue now, I'm not sure if it's related but I can't get OBS to open at all now. Could that have been caused by me reinstalling it?

**Comment by Mythotical** [comment]
> > Oh ok. I have a new issue now, I'm not sure if it's related but I can't get OBS to open at all now. Could that have been caused by me reinstalling it?
> 
> If it's 30.2.0-beta2, then it's a known issue that is fixed in beta3.

LOL It is. Thanks. Guess I'll wait on Beta 3 or go back to 30.1.2 for now.

**Comment by Mythotical** [comment]
Nevermind, I see 3 is released. Thanks for all the help.
